tokens/figma/update-middleware.mjs

```diff
@@ -0,0 +1,796 @@
+          );
+        }
+      }
+    }
+
+    // Step 9: Send the data to update the Skin collection (POST)
+    const updateResponse = await fetch(
+      `https://api.figma.com/v1/files/${FILE_KEY}/variables`,
+      {
+        method: "POST",
+        headers: {
+          "X-Figma-Token": FIGMA_TOKEN,
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(newData),
+      }
+    );
+
+    if (!updateResponse.ok) {
+      const errorText =
+        await updateResponse.text();
+      throw new Error(
+        `Error updating Skin collection: ${updateResponse.statusText}. Response: ${errorText}`
+      );
+    }
+
+    return newData; // Returning newData for debugging
+  } catch (error) {
+    console.error("Error:", error);
+    throw error;
+  }
+}
+
+async function updateSkinOtherVariables(
+  jsonData,
+  brands,
+  FILE_KEY,
+  FIGMA_TOKEN
+) {
+  const response = await fetch(
+    `https://api.figma.com/v1/files/${FILE_KEY}/variables/local`,
+    {
+      method: "GET",
+      headers: {
+        "X-Figma-Token": FIGMA_TOKEN,
+        "Content-Type": "application/json",
+      },
+    }
+  );
+
+  const figmaData = await response.json();
+  const existingVariables =
+    figmaData.meta.variables;
+  const existingCollections =
+    figmaData.meta.variableCollections;
+
+  const newData = {
+    variables: [],
+    variableModeValues: [],
+  };
+
+  const fontFamilies = {
+    movistar: "On Air",
+    "vivo-new": "Vivo type",
+    "o2-new": "On Air",
+    telefonica: "Telefonica Sans",
+    blau: "SF Pro Text",
+    tu: "Telefonica Sans",
+  };
+
+  const iconSets = {
+    movistar: "Default",
+    "vivo-new": "Vivo",
+    "o2-new": "O2",
+    telefonica: "Default",
+    blau: "Blau",
+    tu: "Default",
+  };
+
+  // Loop through each brand to process its specific tokens
+  for (const brand of brands) {
+    const variableGroups = [
+      {
+        variables: jsonData[brand]?.radius || [],
+        collectionName: COLLECTION_NAMES.BRAND,
+        resolvedType: VARIABLE_TYPES.FLOAT,
+        variableScopes: ["CORNER_RADIUS"],
+        hasAlias: false,
+      },
+      {
+        variables:
+          jsonData[brand]?.fontWeight || [],
+        collectionName: COLLECTION_NAMES.BRAND,
+        resolvedType: VARIABLE_TYPES.STRING,
+        variableScopes: ["FONT_WEIGHT"],
+        hasAlias: false,
+      },
+      {
+        variables:
+          jsonData[brand]?.fontSize || [],
+        collectionName: COLLECTION_NAMES.BRAND,
+        resolvedType: VARIABLE_TYPES.FLOAT,
+        variableScopes: ["FONT_SIZE"],
+        hasAlias: false,
+      },
+      {
+        variables:
+          jsonData[brand]?.lineHeight || [],
+        collectionName: COLLECTION_NAMES.BRAND,
+        resolvedType: VARIABLE_TYPES.FLOAT,
+        variableScopes: ["LINE_HEIGHT"],
+        hasAlias: false,
+      },
+      {
+        variables: [
+          {
+            name: "fontFamily/fontFamily",
```

Comment:
I think "fontFamily" is an inconsistency comparing with "Icon Set"

should we change the name to "Font Family"? Or change Icon Set to "iconSet"?

Should we follow a differente nomenclatura for variables that only exist in Figma? or should we keep the same naming convention for all?

Reply:
I've updated Icon Set to be iconSet, all mistica token names are defined in camelCase

Reply:
remember rename the variable manually in Mística Skin Figma file